corporate/lib/stripe.py

```diff
@@ -1920,11 +1929,7 @@ def make_end_of_cycle_updates_if_needed(
             licenses_at_next_renewal = last_ledger_entry.licenses_at_next_renewal
             assert licenses_at_next_renewal is not None
 
-            if (
-                plan.tier == CustomerPlan.TIER_SELF_HOSTED_LEGACY
-                and plan.end_date == next_billing_cycle
-                and plan.status == CustomerPlan.ACTIVE
-            ):
+            if plan.end_date == next_billing_cycle and plan.status == CustomerPlan.ACTIVE:
```

Comment:
@amanagr remind me why we had this logic conditional on the legacy plan tier previously?

I also wonder if this logic is correct in the case that there's a next plan, or if it could result in a problematic intermediate downgrade that isn't intended.

Reply:
If there's a next plan, the status of current_plan would be `SWITCH_PLAN_TIER_AT_PLAN_END`

Reply:
Only reason I had that check because we didn't had plans to support anything like this at the time.